//============================================================================
// DESCRIPTION : Lab Five  C++II ToUpper Inheritance Class, with myStringClass.
// CLASS       : COSC 1437.S02 â€“ Lab 4
// AUTHOR      : Matthew Dowell
// DATE        : 04/06/2015
//============================================================================

#include <iostream>
#include "UCString.h"
#include "DigitString.h"

using namespace std;

int main() {


	UCString A2 ("charlie");

	UCString A3;

	String A4 ("Johnny");

	cout << A2 << endl;

	cout << "Assign = " << endl;
	A3 = A2;

	cout << A3 << endl;

	A3 = A4;

	cout << "I'm assigning something" << endl;
	cout << A3 << endl;


	cout << endl;
	cout << "Concat test" << endl;
	A2 &= A4;
	cout << A2 << endl;


	cout << endl;
	cout << "attempting digit string" << endl;
	try{
	    digitString DStest ("12afc25");
	    // do stuff with f
	}
	catch(invalid_argument& e){
	    cout << "testdigitString class can only accept digits" << endl;
	}

	cout << "Loading DStest2 with digits, and running through Concat, and Assignment tests." << endl;

	try{
		digitString DStest2 ("987654321");
		digitString DStest3;


		DStest3 = DStest2;

		DStest3 = "123456789";

		DStest3 &= DStest2;

	}catch(invalid_argument& e){
	    cout << "testdigitString class can only accept digits" << endl;
	}

	cout << endl;
	cout << "I tried to run" << endl;

/*
 * try{
 *
 * DigitString DS;
 * }
 * catch (WCS_String::Exceptions e){
 * }
 * catch (DigitString::Exceptions e){
 * }
 * catch (...){ //this will catch something, but you can't tell what you've caught.
 * }
 *
 * the compiler should continue along after it receives the catch.
 *
 *
 *
 *
 */

	return 0;
}
